onnxruntime/core/providers/webnn/builders/impl/dynamicQuantizeLinear_op_builder.cc

```diff
@@ -14,34 +14,194 @@
+                               const logging::Logger& logger) const override;
 };
 
-Status DynamicQuantizaLinearOpBuilder::AddToModelBuilderImpl(ModelBuilder& model_builder,
+// DynamicQuantizeLinear is a function defined as follows:
+// DynamicQuantizeLinear (x) => (y, y_scale, y_zero_point)
+// {
+//    Q_Min = Constant <value: tensor = float {0}> ()
+//    Q_Max = Constant <value: tensor = float {255}> ()
+//    X_Min = ReduceMin <keepdims: int = 0> (x)
+//    X_Min_Adjusted = Min (X_Min, Q_Min)
+//    X_Max = ReduceMax <keepdims: int = 0> (x)
+//    X_Max_Adjusted = Max (X_Max, Q_Min)
+//    X_Range = Sub (X_Max_Adjusted, X_Min_Adjusted)
+//    Scale = Div (X_Range, Q_Max)
+//    Min_Scaled = Div (X_Min_Adjusted, Scale)
+//    Initial_ZeroPoint_FP = Sub (Q_Min, Min_Scaled)
+//    Clipped_ZeroPoint_FP = Clip (Initial_ZeroPoint_FP, Q_Min, Q_Max)
+//    Rounded_ZeroPoint_FP = Round (Clipped_ZeroPoint_FP)
+//    Zeropoint = Cast <to: int = 2> (Rounded_ZeroPoint_FP)
+//    y_scale = Identity (Scale) (Skip in WebNN)
+//    y_zero_point = Identity (Zeropoint) (Skip in WebNN)
+//    y = QuantizeLinear (x, Scale, Zeropoint)
+// }
+Status DynamicQuantizeLinearOpBuilder::AddToModelBuilderImpl(ModelBuilder& model_builder,
                                                              const Node& node,
                                                              const logging::Logger& logger) const {
   const auto& input_defs = node.InputDefs();
   emscripten::val input = model_builder.GetOperand(input_defs[0]->Name());
-  emscripten::val output_array;
-  std::vector<int64_t> input_shape;
-  ORT_RETURN_IF_NOT(GetShape(*input_defs[0], input_shape, logger), "Cannot get shape");
-  emscripten::val options = emscripten::val::object();
-  options.set("label", node.Name());
+  emscripten::val common_options = emscripten::val::object();
+
+  // Q_Min = Constant <value: tensor = float {0}> ()
+  emscripten::val q_min = model_builder.CreateOrGetConstant<float>(ONNX_NAMESPACE::TensorProto_DataType_FLOAT, 0.0f);
+  // Q_Max = Constant <value: tensor = float {255}> ()
+  emscripten::val q_max = model_builder.CreateOrGetConstant<float>(ONNX_NAMESPACE::TensorProto_DataType_FLOAT, 255.0f);
+
+  // X_Min = ReduceMin <keepdims: int = 0> (x)
+  common_options.set("label", node.Name() + "_x_min");
+  emscripten::val x_min = model_builder.GetBuilder().call<emscripten::val>("reduceMin", input, common_options);
+
+  // X_Min_Adjusted = Min (X_Min, Q_Min)
+  common_options.set("label", node.Name() + "_x_min_adjusted");
+  emscripten::val x_min_adjusted = model_builder.GetBuilder().call<emscripten::val>("min", x_min, q_min, common_options);
+
+  // X_Max = ReduceMax <keepdims: int = 0> (x)
+  common_options.set("label", node.Name() + "_x_max");
+  emscripten::val x_max = model_builder.GetBuilder().call<emscripten::val>("reduceMax", input, common_options);
+
+  // X_Max_Adjusted = Max (X_Max, Q_Min)
```

Comment:
Confirm not a typo?

```diff
-  // X_Max_Adjusted = Max (X_Max, Q_Min)
+  // X_Max_Adjusted = Max (X_Max, Q_Max)
```

Because it seems weird for the above to have `  // X_Min_Adjusted = Min (X_Min, Q_Min)`.